web/src/components/graph/areagraph.jsx

```diff
@@ -118,7 +118,11 @@ const AreaGraph = React.memo(
     const valueScale = useMemo(() => getValueScale(containerHeight, totalValues), [containerHeight, totalValues]);
     const datetimes = useCurrentZoneHistoryDatetimes();
     const startTime = datetimes.at(0);
-    const endTime = datetimes.at(-1);
+    const lastTime = datetimes.at(-1);
+    const intervalMs = datetimes.length > 1 ? lastTime.getTime() - datetimes.at(-2).getTime() : undefined;
```

Comment:
Arguably this could have been hard coded for each time range to make it a bit easier to know where things break in the future, but it's also nice that it's shared across time ranges.